src/persist-client/src/batch.rs

```diff
@@ -807,48 +819,44 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
                         .inc_by(stats_step_timing.as_secs_f64());
                     stats
                 });
-                (payload_len, stats)
+
+                HollowBatchPart {
+                    key: partial_key,
+                    encoded_size_bytes: payload_len,
+                    key_lower: key_lower.unwrap_or_else(Vec::new),
+                    stats,
+                }
             }
             .instrument(write_span),
         );
-        self.writing_parts.push_back((partial_key, handle));
+        self.writing_parts.push_back(handle);
 
         while self.writing_parts.len() > self.cfg.batch_builder_max_outstanding_parts {
             batch_metrics.write_stalls.inc();
-            let (key, handle) = self
+            let handle = self
                 .writing_parts
                 .pop_front()
                 .expect("pop failed when len was just > some usize");
-            let (encoded_size_bytes, stats) = match handle
+            let part = match handle
                 .instrument(debug_span!("batch::max_outstanding"))
                 .await
             {
                 Ok(x) => x,
-                Err(err) if err.is_cancelled() => (0, None),
```

Comment:
I couldn't figure out why we were swallowing this error, and a git-history dive didn't help. I figured the build would tell me if this was important for stability, but so far it seems happy? Anyways please tell me if I should go back to the old structure... I should be able to preserve it with a little extra work.

Reply:
I've hit this same thing before myself and can't remember either